pymc/variational/approximations.py

```diff
@@ -248,9 +249,7 @@ def randidx(self, size=None):
                 pass
         else:
             size = tuple(np.atleast_1d(size))
-        return self._rng.uniform(
-            size=size, low=pm.floatX(0), high=pm.floatX(self.histogram.shape[0]) - pm.floatX(1e-16)
-        ).astype("int32")
+        return self.srng.integers(size=size, low=0, high=self.histogram.shape[0])
```

Comment:
Better to use `at.random.integers` directly. Automatic updates are handled by `compile_pymc` anyway, which is the only thing that RandomStream really does. Not relying on it avoids issues when cloning graphs (the rng.update graph still refers to the original one) because that's invisible to tools like `clone_replace`: https://github.com/Theano/Theano/issues/1467

Reply:
If you need to figure out the updates (e.g, inside a Scan) you can use this utility:

https://github.com/pymc-devs/pymc/blob/4cac135cd21c24ca085cfce4f27416d061c23a5e/pymc/pytensorf.py#L1063